internal/cli/task_list.go

```diff
@@ -0,0 +1,142 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/jsonpb"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type TaskListCommand struct {
+	*baseCommand
+
+	flagJson  bool
+	flagLimit int
+	flagDesc  bool
+}
+
+func (c *TaskListCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+	); err != nil {
+		return 1
+	}
+	ctx := c.Ctx
+
+	resp, err := c.project.Client().ListTask(ctx, &pb.ListTaskRequest{})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if len(resp.Tasks) == 0 {
+		return 0
+	}
+	tasks := resp.Tasks
+
+	// reverse task list if requested
+	if c.flagDesc {
+		var reverse []*pb.GetTaskResponse
+		for i := len(tasks) - 1; i >= 0; i-- {
+			reverse = append(reverse, tasks[i])
+		}
+		tasks = reverse
```

Comment:
I remember trying this initially when I did list reversal for `waypoint job list` and it didn't turn out like I expected it do 🤔 will take a look for this one though! I didn't need it for job list because I could sort on timestamp.

Reply:
This is a pretty nit-picky / optimization thing so  feel free to simply pass on the suggestion

Reply:
No worries, I just tried it in Go Playground, seems like it should work! I'll update it